src/Sequence.cc

```diff
@@ -282,7 +284,34 @@ Sequence::ConstIterator Sequence::erase(Sequence::ConstIterator begin,
     return return_iter;
 }
 
-void Sequence::execute(Context& context, CommChannel* comm)
+void Sequence::execute(Context& context, CommChannel* comm_channel)
+{
+    handle_execution(context, comm_channel, "Sequence",
+        [this](Context& context, CommChannel* comm)
+        {
+            check_syntax();
+            execute_range(steps_.begin(), steps_.end(), context, comm);
+        });
+}
+
+void Sequence::execute_single_step(Context& context, CommChannel* comm_channel,
+                                   StepIndex step_index)
+{
+    if (step_index >= size())
+        throw Error(cat("Invalid step index ", step_index));
+
+    handle_execution(context, comm_channel, cat("Step ", step_index + 1),
+        [this, step_index](Context& context, CommChannel* comm)
+        {
+            auto step_it = steps_.begin() + step_index;
+            if (executes_script(step_it->get_type()))
+                step_it->execute(context, comm, step_index);
+        });
+}
+
+void Sequence::handle_execution(Context& context, CommChannel* comm,
+                                gul14::string_view exec_block_name,
+                                std::function<void(Context&, CommChannel*)> fct)
```

Comment:
I agree the name could be more descriptive, but `execute` is imho also bad ;^)

(Esp. because the variable is used faaaaaaar below ;) and not in the next few lines.)

That is some kind of `executor`, but we already have that, so maybe `runner`?

Reply:
You are both right, `fct` is a little over-generic. Hmmm... `runner` makes sense to me. Will adapt this accordingly.